iati/core/tests/test_rulesets.py

```diff
@@ -315,41 +351,67 @@ def rule_type(self):
 
     all_valid_cases = [
         {'paths': ['element1']},  # single path
+        {'paths': ['element6/@attribute']},
         {'paths': ['element2', 'element3']},  # multiple paths
+        {'paths': ['element7/@attribute', 'element8/@attribute']},
         {'paths': ['element4', 'element4']},  # duplicate paths
-        {'paths': ['element5/@attribute']},  # single path for attribute
-        {'paths': ['element6/@attribute', 'element7/@attribute']},  # multiple paths for attribute
-        {'paths': ['element8/@attribute', 'element8/@attribute']}
+        {'paths': ['element9/@attribute', 'element9/@attribute']},
+        {'paths': ['element11']}  # multiple identitcal elements
     ]
 
-    @pytest.fixture(params=all_valid_cases)
-    def instantiating_case(self, request):
-        """Permitted case for instatiating this Rule."""
-        return request.param
-
-    @pytest.fixture(params=[
+    uninstantiating_cases = [
         {'paths': []},  # empty path array
+        {'paths': ['']},  # path is an empty string
         {'paths': 'element'},  # non-array `paths`
         {'paths': [3]},  # non-string value in path array
-        {'paths': ['element', 3]},  # mixed string and non-string value in path array
+        {'paths': ['element', 3]},  # mixed string and non-string values in path array
+        {'paths': ['element/@attribute', 3]},
         {},  # empty dictionary
         {'paths': {'element'}},  # dictionary paths
+        {'paths': {'element/@attribute'}},
         {'paths': 'element'},  # non-array `paths`
         {'paths': [3]},  # non-string value in path array
-        {'paths': ['element/@attribute', 3]},  # mixed string and non-string value in path array
-        {},  # empty dictionary
-        {'paths': {'element/@attribute'}}  # dictionary paths
-    ])
+        {}  # empty dictionary
+    ]
+
+    invalidating_cases = [
+        {'paths': ['element1']},  # single path, no matches
+        {'paths': ['element9/@attribute']},
+        {'paths': ['element7', 'element8']},  # multiple paths, both expected matches missing
+        {'paths': ['element13/@attribute', 'element14/@attribute']}
+    ]
+
+    @pytest.fixture(params=all_valid_cases)
```

Comment:
It should be possible for the various fixtures that take the same sort of input and return `request.param` to exist within `RuleSubclassTestBase` so they don't need repeating for each Rule.

Reply:
future branch problem